### PR TITLE
Update tests so we can migrate SlotFill component to the stable context API

### PR DESCRIPTION
**Pull request by oandregal**
This PR prepares our test suite to be testable against the stable Context API, so we can migrate our code safely. Once this lands, we can merge https://github.com/WordPress/gutenberg/pull/11123

### CODE REVIEW

**Comment by nosolosw** [comment]
As per @gziolo suggestion [here](https://github.com/WordPress/gutenberg/pull/11123#pullrequestreview-170222999) I've tried not to use snapshot testing, but having the HTML inside the test. It looks like it's more difficult than it's supposed to and I couldn't figure out.

I can give it a second try if that's a blocker. I'm actually inclined to think that snapshot testing is a fine choice here.

**Comment by gziolo** [comment]
@aduth @nerrad as you have some experience with writing test using `react-test-renderer`. Is there an easy way to replicate the following assertion written with `enzyme`:

```js
expect( element.html() ).toBe( '<div><span></span></div>' );
```

If not let's use snapshots instead. My only concern is that having 10 snapshots for so simple HTML snippets makes those tests much harder to read. However, it doesn't seem to be a blocker here.

**Comment by nerrad** [comment]
If you're just testing a simple structure like that (i.e. no expected css class or props) I would think snapshots are fine. Provided the test has good enough description the snapshots aren't _too_ hard to follow.  As soon as you get to testing unit written components though you have to keep in mind that `toJson` doesn't include them.

**Comment by nosolosw** [comment]
Rebased from master.

**Comment by youknowriad** [comment]
@gziolo 

```
expect( element.html() ).toBe( '<div><span></span></div>' );
```

Why can just do :) 

```
const div = document.createElement( 'div' );
render( something, div );
console.log( div.innerHTML );
```

**Review by youknowriad — APPROVED** [review]
Tests pass, that's good enough for me 👍 

I don't want to block the other PR

**Comment by nosolosw** [comment]
Cool, going to merge both PRs. We can always iterate on how tests are written.
